### ISSUE TITLE
mixed up paths when outputting JSON files

### ISSUE THREAD

**Issue by eighthave** [reddr/LibScout#15]
In the WU run on 1.2TB of APKs, I'm using the JSON output.  It outputs mostly using a directory hierarchy based on the segments of the Java _Package Name_.  But sometimes it outputs straight into the _json/_ dir:

```console
$ ls -1 json/org.mariotaku.twidere*.json json/org/mariotaku/*.json
json/org/mariotaku/org.mariotaku.imageviewergl_1_1.json
json/org/mariotaku/org.mariotaku.twidere_135_135.json
json/org/mariotaku/org.mariotaku.twidere_166_166.json
json/org/mariotaku/org.mariotaku.twidere_181_181.json
json/org/mariotaku/org.mariotaku.twidere_196_196.json
json/org/mariotaku/org.mariotaku.twidere_213_213.json
json/org/mariotaku/org.mariotaku.twidere_260_260.json
json/org/mariotaku/org.mariotaku.twidere_370_370.json
json/org/mariotaku/org.mariotaku.twidere_371_371.json
json/org/mariotaku/org.mariotaku.twidere_379_379.json
json/org.mariotaku.twidere_389_0.json
json/org.mariotaku.twidere_393_0.json
json/org.mariotaku.twidere_394_0.json
```

**Comment by eighthave** [comment]
Also, while we are on the topic of paths, it would be nice if these paths followed the maven-style paths.  Right now, _LibScout_ does not use the last segment of the _Package Name_ in the path.  Maven-style uses all segments.  Right now, _LibScout_ outputs:

```
json/org/mariotaku/failtest_12638458_211.json
json/org/mariotaku/org.mariotaku.imageviewergl_1_1.json
json/org/mariotaku/org.mariotaku.twidere.extension.twitlonger_3_3.json
json/org/mariotaku/org.mariotaku.twidere.extension.twitlonger_4_4.json
json/org/mariotaku/org.mariotaku.twidere.extension.twitlonger_5_5.json
json/org/mariotaku/org.mariotaku.twidere_110_110.json
```
But it would be clearer using Maven-style paths:

```
json/org/mariotaku/imageviewergl/org.mariotaku.imageviewergl_1_1.json
json/org/mariotaku/twidere/failtest_12638458_211.json
json/org/mariotaku/twidere/org.mariotaku.twidere_110_110.json
json/org/mariotaku/twidere/extension/twitlonger/org.mariotaku.twidere.extension.twitlonger_3_3.json
json/org/mariotaku/twidere/extension/twitlonger/org.mariotaku.twidere.extension.twitlonger_4_4.json
json/org/mariotaku/twidere/extension/twitlonger/org.mariotaku.twidere.extension.twitlonger_5_5.json
```

**Comment by reddr** [comment]
Ok, two separate answer to this:

- Why does the output is sometimes straight in the json/ directory?<br>
 A (non-trivially fixable) bug in the Manifest parser causes an exception that prevents LibScout from retrieving the package name/version code. Results are then put in the configured json dir directly (those files end with "_0.json". A good solution would probably be adopting the parsing routines from https://github.com/iBotPeaches/Apktool

- Output following maven-style?<br>
Currently LibScout does not make any assumptions on the apk file names. It only uses the parsed package name to structure output paths. For maven-style either LibScout would require additional information such as group-/artefactid separately or would need to parse it from the app file name that must to follow a maven naming convention.

**Comment by eighthave** [comment]
The code already has the package name since the .json file is named with it.  Why not just always put files into the maven dir structure?  Of course, if the package name is not available, it can go in the json/ root.  I'm speaking about the cases where the package name is available, but there are other things broken about the APK.

As for the maven dir structure, I just mean the directories named after the segments of the package name.  I don't mean the full maven repo file layout. Like you said, artifact ID etc. are not really applicable.

My guess is that LibScout currently makes the dir structure based on the first two segments, e.g. _org.mariotaku.twidere_ and _org.mariotaku.twidere.extension.twitlonger_ both end up in _org/mariotaku/_.  The maven-style dir structure that I mean is just replacing `.` with `/` in the package name and using that as the dir for that file.

**Comment by eighthave** [comment]
just the directory layout based on package name segments is good, that's all I meant by "maven-style paths"

**Comment by reddr** [comment]
fixed in 17c0917bedb882d86ef423ad052050b71cafba13, subpath creation now takes the full package name
